src/lib/custom-deploy/custom-deploy.ts

```diff
@@ -9,15 +11,39 @@ import { trackEventWithEnv } from '../../lib/tracker';
 import { validateDeployFileExt, validateFilename } from '../../lib/validations/custom-deploy';
 
 const DEPLOY_MAX_FILE_SIZE = 4 * GB_IN_BYTES;
+const CUSTOM_DEPLOY_KEY = process.env.CUSTOM_DEPLOY_KEY;
 
 export function isSupportedApp( app: App ): boolean {
 	return WORDPRESS_SITE_TYPE_IDS.includes( app.typeId as number );
 }
 
+export async function validateCustomDeployKey( envId: number ): Promise< void > {
+	if ( CUSTOM_DEPLOY_KEY === '' ) {
```

Comment:
I think you might want to check for false-yness here, since it could be totally undefined
```suggestion
	if ( ! CUSTOM_DEPLOY_KEY ) {
```